qemu/tests/spapr_vty_multi_backends.py

```diff
@@ -0,0 +1,73 @@
+import logging
+import os
+import re
+
+from virttest import error_context
+from virttest import remote
+from virttest import utils_test
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Verify the login guest with multi backends spapr-vty:
+    1) Boot guest with multi spapr-vty with backend
+    2) Modify the kernel cfg file to specify the backend
+    3) for pty and file backend:
+      3.1) open and close chardev
+    4) for unix_socket and tcp_socket
+      4.1) Login guest.
+      4.2) move, create files inside guest
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    prompt = params.get("shell_prompt")
+    vm = env.get_vm(params["main_vm"])
+    vm.wait_for_login()
+    for serial_id in params.objects('serials'):
+        if serial_id != "vs1":
+            hvc_id = int(serial_id.replace('vs', '')) - 1
+            kernel_params = "console=hvc%s,115200" % hvc_id
+            utils_test.update_boot_option(vm, args_added=kernel_params)
+
+        backend = params.object_params(serial_id)["chardev_backend"]
+        serial_device = vm.devices.get(serial_id)
+        chardev_qid = serial_device.get_param("chardev")
+        chardev_device = vm.devices.get_by_qid(chardev_qid)[0]
+
+        logging.info("The currently tested backend is %s." % backend)
+        if backend == 'unix_socket':
+            session = vm.wait_for_serial_login(timeout=60)
+            session.cmd(params['create_file_and_copy'])
+            session.close()
+        elif backend == 'tcp_socket':
+            session = remote.remote_login(client='nc',
+                                          host=chardev_device.params['host'],
+                                          port=chardev_device.params['port'],
+                                          username='root',
+                                          password=params['password'],
+                                          prompt=prompt,
+                                          timeout=240)
+            session.cmd(params['create_file_and_copy'])
+            session.close()
+        elif backend == 'pty':
+            chardev_info = vm.monitor.human_monitor_cmd('info chardev')
+            hostfile = re.findall('%s: filename=pty:(/dev/pts/\\d)?' %
+                                  serial_id, chardev_info)[0]
```

Comment:
An exception will be raised when the pattern is not found because you add `[0]`

Reply:
Corrected